### ISSUE TITLE
Operationalize this repo

### ISSUE THREAD

**Issue by kingdonb** [kingdonb/link-checker-gpt#15]
There is a README for the link checker here, it has not published any releases yet:
https://github.com/kingdonb/link-checker-gpt#link-checker-gpt

An example use is configured as an internal test here - https://github.com/kingdonb/link-checker-gpt/blob/main/.github/workflows/pull-request.yml we can look at this in a separate issue for inclusion in the website CI, it will tell us when we're regressing in a PR. (I don't have another Netlify site that I can test it with, but it runs with a fixed PR number)

We need to test it on a Netlify site, (or we could build an adapter for something else to approximate the Netlify site... hint hint)

**Comment by kingdonb** [comment]
To publish this as a release, we should be building into a container image. That way we can avoid pulling the ruby runtime and gems into the workspace at runtime, and just launch the workflow instantly.

**Comment by kingdonb** [comment]
The logging has been cleaned up in #13 and we can see it works reliably now.

All that remains is to convert this into a so-called "reusable workflow"

https://github.com/orgs/community/discussions/36861

In this transition, we will have to curb our use of external workflows. This means the upload-artifact workflow goes away, and all output will be easier for users to observe directly in the report.

There is no need for a CSV output when the report is properly implemented. We can provide the output clear enough in the workflow logs, if they don't want to use upload-artifact. I am not sure how we can go on providing upload-artifact in the reusable workflow, but it is not a requirement and anyway this change will help to resolve:

* #14

**Comment by kingdonb** [comment]
The v1-beta branch is failing because of:

* #10 

It is otherwise successful here though,

* https://github.com/fluxcd/website/actions/runs/5929294738/job/16076673194?pr=1630

🎉 

Little bit more work to do before we can call this a taggable release

**Comment by kingdonb** [comment]
Fixed #10 via v1-beta

https://github.com/kingdonb/link-checker-gpt/pull/18#issuecomment-1686785221
